R/pkg/R/DataFrame.R

```diff
@@ -594,6 +594,35 @@ setMethod("repartition",
             dataFrame(sdf)
           })
 
+#' RepartitionByColumn
+#'
+#' Return a new SparkDataFrame which has as many partitions as the number of unique
+#' groups identified by column(s) values which are being specified by the input.
+#'
+#' @param x A SparkDataFrame
+#' @param col The column by which the partitioning will be performed
+#'
+#' @family SparkDataFrame functions
+#' @rdname repartition
+#' @name repartition
+#' @export
+#' @examples
+#'\dontrun{
+#' sc <- sparkR.init()
+#' sqlContext <- sparkRSQL.init(sc)
+#' path <- "path/to/file.json"
+#' df <- read.json(sqlContext, path)
+#' newDF <- repartitionByColumn(df, df$col1, df$col2)
+#'}
+setMethod("repartitionByColumn",
```

Comment:
Hi @felixcheung ,

thanks for your prompt response. 
That was my first try too, however there already exist a definition of repartition and if I try the following:
```
setGeneric("repartition",
           function(x, col, ...) {
             standardGeneric("repartition")
          })
```

It fails saying:
  unused argument (numPartitions = c("numeric", ""))
Error : unable to load R code in package ‘SparkR’